src/xnnpack/prefetch.h

```diff
@@ -6,7 +6,37 @@
 #pragma once
 
 #ifdef _MSC_VER
-  #include <intrin.h>
+  #include <intrin.h>  
+#endif
+
+#ifdef __hexagon__
+  #include <hexagon_protos.h>
+
+  // Use fetchl2 at least several hundred cycles prior to using the data.
+  
+  // height: the number of 128 bytes to fetch.
+  // width: we set 128 bytes here.
+  // stride: we set 128 bytes here.
+  XNN_INLINE static void xnn_prefetch_to_l2_linear(const void *address, uint32_t height)
+  {
+    uint32_t info = ((128 << 16) | ((128 << 8) & 0xffff) | (height & 0xffff));
```

Comment:
This doesn't look right. The bits of h1 and h2 overlap the height.

It looks like you are trying to emulate the macro you were using earlier. Why not just use that? 

More generally, I think maybe we should not do this unless you can run on device and measure a performance improvement, so we know this is working as intended, since it's difficult to test prefetching otherwise.

Reply:
For box prefetch, I can use the macro again - 
`uint64_t info = HEXAGON_V64_CREATE_H(direction, stride, width, height);`

For linear prefetch, I need to prepare info into having
0-7: height
8-15: width
16-32: stride
There is no macro defined to prepare this particular dataset, so I can do like this:
```
// set width (128) and height and put into the lower16 bytes.
uint32_t info =(128 << 8) & 0xffff | (height & 0xffff); 

// put stride (128) into the upper 16 bytes.
info =  HEXAGON_V32_CREATE_H(128, info);
```

Yes it would make more sense with performance measurement as I cannot tell if the prefetcher works or not from simulator. Later we also need to make sure to have input and weight is aligned to have prefetch works well.

Reply:
Why not `HEXAGON_V64_CREATE_H(0, 0, size_bytes, 1)`?

Reply:
I guess the last parameter '1' is the height? 

And yes we can use that macro but, shouldn't we still need `(128 << 8) & 0xffff) | (height & 0xffff);` for the lower half-word? Otherwise `width` will be 0 but stride is set to `size_bytes`. If having stride correctly set is sufficient, then we can use `HEXAGON_V64_CREATE_H(0, 0, size_bytes, height)`.

Reply:
Why is there a height at all? Aren't we trying to describe an N x 1 rectangle, i.e. a single row? The stride doesn't matter, and height is 1? 

Doesn't this macro handle all of the bit shuffling for you?

Reply:
Ahh, I assumed `size_bytes` as 128 bytes and `height` as the number of lines of 128 bytes. You meant `size_bytes = BATCH_TILE * sizeof(float)` correct? Then using your method makes perfect sense. And yes confirmed from the manual that `width` is the important part and `stride` doesn't matter. I can change in this way and we can come back and test the performance. Please let me know if I still misunderstood!